### PR TITLE
Remove fall back to legacy type mapping in Redshift

### PR DESCRIPTION
**Pull request by ssheikin**
We should not fall back to legacy behavior,
the mappings should be explicit (the legacyToWriteMapping is just a
copy of some generic default mappings that used to exist)

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(x) Release notes are required, with the following suggested text:
Remove fall back to legacy type mapping in Redshift.

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by kokosing — APPROVED** [review]

File: plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftConnectorTest.java [hunk 1]
```diff
@@ -101,6 +103,40 @@ protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
...
+                                   ]
+                                }')
+                                """,
+                        "4, 4"))) {
+            assertQuery("SELECT * FROM " + table.getName(), "VALUES (1), (2), (3), (4)");
+            assertQuery(convertToVarchar, "SELECT * FROM " + table.getName(), """
+                    VALUES
+                    (1, null),
+                    (2, '\"super value string\"'),
+                    (3, '{"r_nations":[{"n_comment":"s. ironic, unusual asymptotes wake blithely r","n_nationkey":16,"n_name":"MOZAMBIQUE"}]}'),
```
Comment (findepi) [inline plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftConnectorTest.java:134]: > Test that Redshift super type is unsupported

it looks "supported", as this test is exercising reading from that type

rename to eg "Test reading Redshift \`super\` type as varchar"

File: plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java [hunk 2]
```diff
@@ -598,8 +598,10 @@ public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connect
                         longTimestampWithTimeZoneWriteFunction()));
         }
 
-        // Fall back to default behavior
-        return legacyToColumnMapping(session, type);
```
[the remark this answers is not in the document]
Comment (ssheikin) [reply plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java:602]: I'm not sure. I don't know why it was introduced and I don't know what deprecation process does trino have.

[the remark this answers is not in the document]
Comment (ssheikin) [reply plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java:602]: Not exactly. Right now if someone would like to use legacy, they can switch `redshift.use-legacy-type-mapping` and get it back. This fallback was enhancing `proper type mapping` with some legacy wrong behaviour.

[the remark this answers is not in the document]
Comment (findepi) [reply plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java:602]: Agreed, this is not exactly same.

When do we plan to remove `redshift.use-legacy-type-mapping`?

[the remark this answers is not in the document]
Comment (ssheikin) [reply plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java:602]: @findepi I'd be glad to remove that code and to answer your questions, but unfortunately I don't know.
I see that this property was added in the commit 
```
Add config property to disable new Redshift type mappings
6fc0b247 Dain Sundstrom <dain@iq80.com> on 23/12/2022 at 00:57
```
According to https://trino.io/development/process.html#contribution-process 
> unless the change is trivial, for example a spelling fix in the documentation

it worth to
> Start a discussion by creating a GitHub [issue](https://github.com/trinodb/trino/issues), or asking on [Slack](https://trino.io/slack.html) 

Also I'd prefer to merge this PR as soon as possible.

File: plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftConnectorTest.java [hunk 3]
```diff
@@ -101,6 +103,40 @@ protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
         }
     }
 
+    @Test
+    public void testSuperColumnType()
```
Comment (hashhar) [inline plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftConnectorTest.java:107]: reminder that CI doesn't run this test class at all (not even with /test-with-secrets).

**Review by hashhar — APPROVED** [review]
I've verified the Redshift tests pass with these changes externally.

**Comment by ssheikin** [comment]
@kokosing @hashhar @wendigo @findepi please merge

**Pull request merged**